react-native/react/native/remote-manager.desktop.js

```diff
@@ -129,28 +143,67 @@ class RemoteManager extends Component {
     })
   }
 
+  showUpdatePromptComponents () {
+    const {showUpdateState} = this.props
+
+    if (showUpdateState.closed) {
+      return
+    }
+
+    const windowOpts = {
+      width: 480, height: 685 + 20 /* TEMP workaround for header mouse clicks in osx */,
+      resizable: true,
+      fullscreen: false,
+      show: false,
+      frame: false
+    }
+
+    return (
+      <RemoteComponent
+        windowsOpts={windowOpts}
+        waitForState
+        component='update'
+        substore='update'
+        onCancel={() => this.props.updateOnCancel()}
+        onSkip={() => this.props.updateOnSkip()}
+        onSnooze={() => this.props.updateOnSnooze()}
+        onUpdate={() => this.props.updateOnUpdate()}
+        onRemoteClose={() => this.props.updateOnCancel()}
+        setAlwaysUpdate={alwaysUpdate => this.props.setAlwaysUpdate(alwaysUpdate)}
+      />
+    )
+  }
+
   render () {
     return (
       <div>
         {Object.keys(this.state.popups).filter(username => !this.props.trackers[username].closed).map(username => this.state.popups[username])}
         {this.pinentryRemoteComponents()}
+        {this.showUpdatePromptComponents()}
       </div>
     )
   }
 }
 
 RemoteManager.propTypes = {
-  pinentryOnCancel: React.PropTypes.any,
-  pinentryOnSubmit: React.PropTypes.any,
-  registerPinentryListener: React.PropTypes.any,
-  registerIdentifyUi: React.PropTypes.any,
+  pinentryOnCancel: React.PropTypes.func,
```

Comment:
So you totally can move these into the components themselves and that'll normally work.

The problem with this specific example is that these actions are relying on a piece of state that exists only in the main window (`updateResponse` https://github.com/keybase/client/pull/1544/files#diff-df6b791fcfad6c76f1f5ac2a4090a827R45)

A better solution I think would be to have actions take in `updateResponse` as an argument. Then the actions can be bound through `connect` like usual (through `mapDispatchToProps`). And you would only have to pass `updateResponse` to the remote component.

You could even use `ownProps` in the `mapDispatchToProps` to prefill the `updateResponse` for the actions.

lmk if I could explain this better, but that should really clean all this stuff up :)